eksctl update cluster --config-file=config.yaml panic

**What happened?**
```
eksctl update cluster --config-file=config.yaml
```
When I try to update cluster and I'm getting  a panic message
```
[ℹ]  using region us-west-2
[!]  NOTE: config file is used for finding cluster name and region
[!]  NOTE: cluster VPC (subnets, routing & NAT Gateway) configuration changes are not yet implemented
[ℹ]  re-building cluster stack "eksctl-k8s-prd-cluster"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2162353]

goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).addOutputsForVPC(0xc0008cc0f0)
        /src/pkg/cfn/builder/vpc.go:160 +0x103
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).AddAllResources(0xc0008cc0f0, 0xc0008cc0f0, 0xc00067b858)
        /src/pkg/cfn/builder/cluster.go:49 +0xac
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).AppendNewClusterStackResource(0xc00089faa0, 0xc00058f301, 0xc0004d8600, 0xc00089faa0, 0x38e1380)
        /src/pkg/cfn/manager/cluster.go:86 +0x6c4
github.com/weaveworks/eksctl/pkg/ctl/update.doUpdateClusterCmd(0xc000276e80, 0xb, 0xc0000898c0)
        /src/pkg/ctl/update/cluster.go:101 +0x344
github.com/weaveworks/eksctl/pkg/ctl/update.updateClusterCmd.func1(0xc00022a900, 0xc000669ce0)
        /src/pkg/ctl/update/cluster.go:22 +0x2a
github.com/weaveworks/eksctl/pkg/ctl/cmdutils.run(0xc000300dc0)
        /src/pkg/ctl/cmdutils/cmd.go:105 +0x2b
github.com/weaveworks/eksctl/pkg/ctl/cmdutils.(*Cmd).SetRunFuncWithNameArg.func1(0xc000357180, 0xc000301990, 0x0, 0x1)
        /src/pkg/ctl/cmdutils/cmd.go:100 +0x83
github.com/spf13/cobra.(*Command).execute(0xc000357180, 0xc000301970, 0x1, 0x1, 0xc000357180, 0xc000301970)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0xc00016cc80, 0xc0003018f0, 0x1, 0x1)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:850 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:800
main.main()
        /src/cmd/eksctl/main.go:71 +0x3fc
```

**How to reproduce it?**
Running this command:
```
envsubst <config.template.yaml > config.yaml
eksctl update cluster --config-file=config.yaml
```
config.template.yaml
```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: ${CLUSTER}
  region: us-west-2
  version: "${EKS_VERSION}"

vpc:
  subnets:
    private:
      us-west-2a:
        id: ${PRIVATE_SUBNET1}
      us-west-2b: 
        id: ${PRIVATE_SUBNET2}
      us-west-2c: 
        id: ${PRIVATE_SUBNET3}
```

**Anything else we need to know?**
I using codebuild to create/update the cluster with IAM Role

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.4.0"} 
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.7", GitCommit:"6f482974b76db3f1e0f5d24605a9d1d38fad9a2b", GitTreeState:"clean", BuildDate:"2019-03-29T16:15:10Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"} 
```

**Logs**
```
eksctl update cluster --config-file=config.yaml -v 4
2019-08-19T15:09:31+03:00 [ℹ]  using region us-west-2
2019-08-19T15:09:32+03:00 [▶]  role ARN for the current session is 
2019-08-19T15:09:33+03:00 [▶]  cluster = {
  Arn: "",
  CertificateAuthority: {
    Data: ""
  },
  CreatedAt: 2019-08-19 08:23:12 +0000 UTC,
  Endpoint: "",
  Logging: {
    ClusterLogging: [{
        Enabled: false,
        Types: [
          "api",
          "audit",
          "authenticator",
          "controllerManager",
          "scheduler"
        ]
      }]
  },
  Name: "k8s-prd",
  PlatformVersion: "eks.2",
  ResourcesVpcConfig: {
    EndpointPrivateAccess: false,
    EndpointPublicAccess: true,
    SecurityGroupIds: ["sg-12345678"],
    SubnetIds: ["subnet-1","subnet-2","subnet-3"],
    VpcId: "vpc-1"
  },
  RoleArn: "",
  Status: "ACTIVE",
  Version: "1.13"
}
2019-08-19T15:09:33+03:00 [!]  NOTE: config file is used for finding cluster name and region
2019-08-19T15:09:33+03:00 [!]  NOTE: cluster VPC (subnets, routing & NAT Gateway) configuration changes are not yet implemented
2019-08-19T15:09:37+03:00 [▶]  cfg.json = \
{
    "kind": "ClusterConfig",
    "apiVersion": "eksctl.io/v1alpha5",
    "metadata": {
        "name": "k8s-prd",
        "region": "us-west-2",
        "version": "1.13"
    },
    "iam": {},
    "vpc": {
        "id": "vpc-1",
        "cidr": "10.81.48.0/22",
        "securityGroup": "sg-12345678",
        "subnets": {
            "private": {
                "us-west-2a": {
                    "id": "subnet-1",
                    "cidr": "10.69.80.0/23"
                },
                "us-west-2b": {
                    "id": "subnet-2",
                    "cidr": "10.69.82.0/23"
                },
                "us-west-2c": {
                    "id": "subnet-3",
                    "cidr": "10.69.84.0/23"
                }
            }
        },
        "sharedNodeSecurityGroup": "sg-12345678"
    },
    "nodeGroups": [
        {
            "name": "KubeSystem",
            "ami": "ami-12345678",
            "amiFamily": "AmazonLinux2",
            "instanceType": "m5.xlarge",
            "privateNetworking": true,
            "securityGroups": {
                "withShared": true,
                "withLocal": true
            },
            "desiredCapacity": 1,
            "volumeSize": null,
            "volumeType": "gp2",
            "volumeIOPS": null,
            "labels": {
                "role": "kubesystem"
            },
            "ssh": {
                "allow": false
            },
            "iam": {
                "withAddonPolicies": {
                    "imageBuilder": false,
                    "autoScaler": false,
                    "externalDNS": false,
                    "certManager": false,
                    "appMesh": null,
                    "ebs": false,
                    "fsx": false,
                    "efs": false,
                    "albIngress": false,
                    "xRay": false,
                    "cloudWatch": false
                }
            }
        }
    ],
    "availabilityZones": [
        "us-west-2c",
        "us-west-2a",
        "us-west-2b"
    ],
    "status": {
        "endpoint": "",
        "certificateAuthorityData": "",
        "arn": ""
    }
}

2019-08-19T15:09:37+03:00 [ℹ]  re-building cluster stack "eksctl-k8s-prd-cluster"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2162353]

goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).addOutputsForVPC(0xc00096a4b0)
        /src/pkg/cfn/builder/vpc.go:160 +0x103
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).AddAllResources(0xc00096a4b0, 0xc00096a4b0, 0xc000509858)
        /src/pkg/cfn/builder/cluster.go:49 +0xac
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).AppendNewClusterStackResource(0xc000ae4210, 0xc000161901, 0xc000582680, 0xc000ae4210, 0x38e1380)
        /src/pkg/cfn/manager/cluster.go:86 +0x6c4
github.com/weaveworks/eksctl/pkg/ctl/update.doUpdateClusterCmd(0xc000583280, 0xb, 0xc000146000)
        /src/pkg/ctl/update/cluster.go:101 +0x344
github.com/weaveworks/eksctl/pkg/ctl/update.updateClusterCmd.func1(0xc00012f600, 0xc00069bce0)
        /src/pkg/ctl/update/cluster.go:22 +0x2a
github.com/weaveworks/eksctl/pkg/ctl/cmdutils.run(0xc0002f6ab0)
        /src/pkg/ctl/cmdutils/cmd.go:105 +0x2b
github.com/weaveworks/eksctl/pkg/ctl/cmdutils.(*Cmd).SetRunFuncWithNameArg.func1(0xc000158c80, 0xc0003a80f0, 0x0, 0x3)
        /src/pkg/ctl/cmdutils/cmd.go:100 +0x83
github.com/spf13/cobra.(*Command).execute(0xc000158c80, 0xc0003a8060, 0x3, 0x3, 0xc000158c80, 0xc0003a8060)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0xc00034bb80, 0xc0002f70c0, 0x1, 0x1)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:850 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:800
main.main()
        /src/cmd/eksctl/main.go:71 +0x3fc
```

--- a/pkg/apis/eksctl.io/v1alpha5/types.go
+++ b/pkg/apis/eksctl.io/v1alpha5/types.go
@@ -366,13 +366,12 @@ func NewClusterConfig() *ClusterConfig {
 // NewClusterVPC creates new VPC config for a cluster
 func NewClusterVPC() *ClusterVPC {
 	cidr := DefaultCIDR()
-	nat := DefaultClusterNAT()
 
 	return &ClusterVPC{
 		Network: Network{
 			CIDR: &cidr,
 		},
-		NAT:              nat,
+		NAT:              DefaultClusterNAT(),
 		AutoAllocateIPv6: Disabled(),
 	}
 }

--- a/pkg/cfn/manager/cluster.go
+++ b/pkg/cfn/manager/cluster.go
@@ -63,17 +63,14 @@ func (c *StackCollection) AppendNewClusterStackResource(plan bool) (bool, error)
 
 	// NOTE: currently we can only append new resources to the stack,
 	// as there are a few limitations:
-	// - it must work with VPC that are imported as well as VPC that
-	//   is mamaged as part of the stack;
+	// - it must work with VPC that is imported as well as VPC that
+	//   is managed as part of the stack;
 	// - CloudFormation cannot yet upgrade EKS control plane itself;
 
 	currentTemplate, err := c.GetStackTemplate(name)
 	if err != nil {
 		return false, errors.Wrapf(err, "error getting stack template %s", name)
 	}
-
-	addResources := []string{}
-	addOutputs := []string{}
 
 	currentResources := gjson.Get(currentTemplate, resourcesRootPath)
 	currentOutputs := gjson.Get(currentTemplate, outputsRootPath)
@@ -112,6 +109,12 @@ func (c *StackCollection) AppendNewClusterStackResource(plan bool) (bool, error)
 		currentTemplate, iterErr = sjson.Set(currentTemplate, path, value.Value())
 		return iterErr == nil
 	}
+
+	var (
+		addResources []string
+		addOutputs   []string
+	)
+
 	newResources.ForEach(func(k, v gjson.Result) bool {
 		return iterFunc(&addResources, resourcesRootPath, currentResources, k, v)
 	})

--- a/pkg/ctl/create/nodegroup.go
+++ b/pkg/ctl/create/nodegroup.go
@@ -83,7 +83,7 @@ func doCreateNodeGroups(cmd *cmdutils.Cmd, updateAuthConfigMap bool) error {
 		return err
 	}
 
-	if err := ctl.GetClusterVPC(cfg); err != nil {
+	if err := ctl.LoadClusterVPC(cfg); err != nil {
 		return errors.Wrapf(err, "getting VPC configuration for cluster %q", cfg.Metadata.Name)
 	}
 

--- a/pkg/ctl/delete/cluster.go
+++ b/pkg/ctl/delete/cluster.go
@@ -123,7 +123,7 @@ func doDeleteCluster(cmd *cmdutils.Cmd) error {
 
 		tasks, err := stackManager.NewTasksToDeleteClusterWithNodeGroups(cmd.Wait, func(errs chan error, _ string) error {
 			logger.Info("trying to cleanup dangling network interfaces")
-			if err := ctl.GetClusterVPC(cfg); err != nil {
+			if err := ctl.LoadClusterVPC(cfg); err != nil {
 				return errors.Wrapf(err, "getting VPC configuration for cluster %q", cfg.Metadata.Name)
 			}
 

--- a/pkg/ctl/update/cluster.go
+++ b/pkg/ctl/update/cluster.go
@@ -83,12 +83,12 @@ func doUpdateClusterCmd(cmd *cmdutils.Cmd) error {
 	case api.Version1_13:
 		cfg.Metadata.Version = api.Version1_13
 	default:
-		// version of control is not known to us, maybe we are just too old...
+		// version of control plane is not known to us, maybe we are just too old...
 		return fmt.Errorf("control plane version %q is not known to this version of eksctl, try to upgrade eksctl first", currentVersion)
 	}
 	versionUpdateRequired := cfg.Metadata.Version != currentVersion
 
-	if err := ctl.GetClusterVPC(cfg); err != nil {
+	if err := ctl.LoadClusterVPC(cfg); err != nil {
 		return errors.Wrapf(err, "getting VPC configuration for cluster %q", cfg.Metadata.Name)
 	}
 

--- a/pkg/eks/eks.go
+++ b/pkg/eks/eks.go
@@ -84,8 +84,8 @@ func (c *ClusterProvider) ControlPlaneVersion() string {
 	return *c.Status.cachedClusterInfo.Version
 }
 
-// GetClusterVPC retrieves the VPC configuration
-func (c *ClusterProvider) GetClusterVPC(spec *api.ClusterConfig) error {
+// LoadClusterVPC loads the VPC configuration
+func (c *ClusterProvider) LoadClusterVPC(spec *api.ClusterConfig) error {
 	stack, err := c.NewStackManager(spec).DescribeClusterStack()
 	if err != nil {
 		return err

--- a/pkg/vpc/vpc.go
+++ b/pkg/vpc/vpc.go
@@ -90,9 +90,9 @@ func describe(povider api.ClusterProvider, vpcID string) (*ec2.Vpc, error) {
 // is treated as the source of truth
 func UseFromCluster(provider api.ClusterProvider, stack *cfn.Stack, spec *api.ClusterConfig) error {
 	if spec.VPC == nil {
-		spec.VPC = &api.ClusterVPC{}
+		spec.VPC = api.NewClusterVPC()
 	}
-	// this call is authoritive, and we can safely override the
+	// this call is authoritative, and we can safely override the
 	// CIDR, as it can only be set to anything due to defaulting
 	spec.VPC.CIDR = nil
 

